### PR TITLE
doc: add OS support for ScyllaDB 5.3

### PR DESCRIPTION
**Pull request by annastuchlik**
Fixes https://github.com/scylladb/scylladb/issues/14084

This commit adds OS support for version 5.3 to the table on the OS Support by Linux Distributions and Version page.

### CODE REVIEW

**Comment by annastuchlik** [comment]
@roydahan @yaronkaikov Please review the row for 5.3.

@tzach I'd like to remove the rows for the outdated versions 2.2, 2.3, and 3.x. In this way, we'll be able to get rid of the columns for Ubuntu 14 and Debian 8 - the table will soon become to wide to fit on the page, and the support for the outdated version will still be part of the 5.1 and 5.2 docs.
Are you OK with that?

**Comment by tzach** [comment]
> @tzach I'd like to remove the rows for the outdated versions 2.2, 2.3, and 3.x. In this way, we'll be able to get rid of the columns for Ubuntu 14 and Debian 8 - the table will soon become to wide to fit on the page, and the support for the outdated version will still be part of the 5.1 and 5.2 docs.
> Are you OK with that?
> 

Yes

**Review by yaronkaikov — APPROVED** [review]

**Comment by annastuchlik** [comment]
@scylladb/scylla-maint Please merge.

File: docs/getting-started/os-support.rst [hunk 1]
```diff
@@ -27,38 +27,34 @@ ScyllaDB Open Source
...
-|   4.1                      | |x|  | |v|  | |v|  | |x|  | |x|  | |x|  | |v|  | |v|   | |x|   | |v|     | |v|   |
-+----------------------------+------+------+------+------+------+------+------+-------+-------+---------+-------+
-|   4.0                      | |x|  | |v|  | |v|  | |x|  | |x|  | |x|  | |v|  | |x|   | |x|   | |v|     | |x|   |
-+----------------------------+------+------+------+------+------+------+------+-------+-------+---------+-------+
-|   3.x                      | |x|  | |v|  | |v|  | |x|  | |x|  | |x|  | |v|  | |x|   | |x|   | |v|     | |x|   |
-+----------------------------+------+------+------+------+------+------+------+-------+-------+---------+-------+
-|   2.3                      | |v|  | |v|  | |v|  | |x|  | |x|  | |v|  | |v|  | |x|   | |x|   | |v|     | |x|   |
-+----------------------------+------+------+------+------+------+------+------+-------+-------+---------+-------+
-|   2.2                      | |v|  | |v|  | |x|  | |x|  | |x|  | |v|  | |x|  | |x|   | |x|   | |v|     | |x|   |
-+----------------------------+------+------+------+------+------+------+------+-------+-------+---------+-------+
```
Comment (roydahan) [inline docs/getting-started/os-support.rst:61]: Why do we list all these old scylla releases, it makes it unclear if you ask me.
Also, why do we list old OS that we don't support anymore.

**Comment by annastuchlik** [comment]
@scylladb/scylla-maint Please backport to branch-5.3. 
This PR updates support for version 5.3.

**Comment by denesb** [comment]
> @scylladb/scylla-maint Please backport to branch-5.3. This PR updates support for version 5.3.

Backport queued :point_up:.

**Pull request merged**